[sanitizer] Fix type in some Min() calls

stefan-sf-ibm opened the pull request:
This is a follow-up to 6dec33834d1fd89f16e271dde9607c1de9554144 and #116957 and #119114.

stefan-sf-ibm commented:
Ping

arichardson commented on compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:
```diff
@@ -532,8 +532,8 @@ INTERCEPTOR(int, strncmp, const char *s1, const char *s2, usize size) {
-  COMMON_INTERCEPTOR_READ_RANGE((ctx), (s1), Min(i1 + 1, size));
-  COMMON_INTERCEPTOR_READ_RANGE((ctx), (s2), Min(i2 + 1, size));
+  COMMON_INTERCEPTOR_READ_RANGE((ctx), (s1), Min<uptr>(i1 + 1, size));
+  COMMON_INTERCEPTOR_READ_RANGE((ctx), (s2), Min<uptr>(i2 + 1, size));
```
IMO it would be cleaner if all of these uptr variables in this function were `usize` since they are sizes and not uintptr_t ("union of pointer and integer").

Would also result in marginally better code generation for CHERI and allows dropping the explicit template argument to min calls.

stefan-sf-ibm replied:
I agree. In the first PR #116957 I did that which got partially merged and then followed the style in PR #119114. I came up with an updated version which works fine for me.

arichardson reviewed (changes requested):
Thanks for working on cleaning this up. Many years ago I did something similar in the downstream CHERI fork but never got around to upstreaming since I thought we were the only architecture where it mattered. 

Could you please change the type of the local variables instead? With that change this LGTM.

stefan-sf-ibm committed:
[sanitizer] Fix type in some Min() calls

This is a follow-up to 6dec33834d1fd89f16e271dde9607c1de9554144
and #116957 and #119114.
